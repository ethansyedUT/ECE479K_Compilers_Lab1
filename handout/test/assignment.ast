#7
_program
  #7
  _class
    Main
    Object
    "assignment.cl"
    (
    #6
    _method
      main
      Int
      #6
      _block
        #5
        _let
          x
          Int
          #3
          _int
            2
          : Int
          #5
          _let
            y
            Bool
            #3
            _no_expr
            : _no_type
            #5
            _assign
              x
              #4
              _int
                79
              : Int
            : Int
          : Int
        : Int
      : Int
    )
