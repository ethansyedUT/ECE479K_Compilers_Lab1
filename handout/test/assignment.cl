class Main {
    main() : Int {{
        (let x : Int <- 2, y : Bool in
           let x : Int in
            x <- 2;
        );
    }};
};