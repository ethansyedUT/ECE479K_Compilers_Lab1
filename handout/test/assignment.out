Main_main() returned 79
